src/message.ts

```diff
@@ -59,18 +59,72 @@ export function encryptStringSync(
 
 // Decryption --
 
+/**
+ * @deprecated
+ *
+ * Causes stack errors on large strings, use {@link parseCloakedString} instead.
+ */
 export const cloakedStringRegex =
   /^v1\.aesgcm256\.(?<fingerprint>[0-9a-fA-F]{8})\.(?<iv>[a-zA-Z0-9-_]{16})\.(?<ciphertext>[a-zA-Z0-9-_]{22,})={0,2}$/
 
+/**
+ * Tests if the input string consists only of URL-safe Base64 chars
+ * (e.g. using `-` and `=` instead of `+` and `/`), and is padded with `=`.
+ *
+ * @returns `true` if the string is a valid URL-safe Base64, else `false`.
+ *
+ * Adapted from <https://github.com/validatorjs/validator.js/blob/ebcca98232399b8404ca6b0ec842ab4596329d58/validator.js#L836-L845>
+ * @license MIT
+ * @copyright Copyright (c) 2016 Chris O'Hara <cohara87@gmail.com>
+ */
+function isBase64(str: string) {
+  const len = str.length
+  if (len % 4 === 0 && !/(^[a-zA-Z0-9-_=])/.test(str)) {
+    return false
+  }
+  const firstPaddingChar = str.indexOf('=')
+  return (
+    firstPaddingChar === -1 ||
+    firstPaddingChar === len - 1 ||
+    (firstPaddingChar === len - 2 && str[len - 1] === '=')
+  )
+}
+
+export function parseCloakedString(input: CloakedString) {
+  const [version, algorithm, fingerprint, iv, ciphertext, nothing] =
+    input.split('.')
+
+  const isCloakedString =
+    version === 'v1' &&
+    algorithm === 'aesgcm256' &&
+    /^[0-9a-fA-F]{8}$/.test(fingerprint) &&
```

Comment:
**suggestion**: Would a case-insensitive Regexp be more performant here?

```suggestion
    /^[0-9a-f]{8}$/i.test(fingerprint) &&
```

Reply:
> LGTM, thanks for the extra care of adding tests and following SemVer!

No worries! And it's cool to see somebody else using https://conventionalcomments.org/ too for PR reviews!

> **suggestion**: Would a case-insensitive Regexp be more performant here?

:shrug: 

I just copied the existing regex in https://github.com/47ng/cloak/blob/52042f33762548431e7f2f5722d6d3758e0bf4ff/src/message.ts#L68

My gut feeling is that it might slow things down, since you'd need to run case-mapping on the entire input string, but it's hard to say without bench-marking it, since CPU caches + pipe-lining means many of these micro-performance enhancements can be very unintuitive.

Considering that the `fingerprint` is only 8 chars, I don't think it matters much.

However, it might make sense to update the `base64` regex, since https://github.com/validatorjs/validator.js/blob/ebcca98232399b8404ca6b0ec842ab4596329d58/validator.js#L107 is using `/i`, and that regex could run on really big strings. Then, we can update all the other regrets[^1] for consistency.

[^1]: there's an old joke saying "the plural of regex is regrets" :wink:

Reply:
Yes, I just commited (and merged) the two 'regrets' in case-insensitive. Can you try and patch`@47ng/cloak@1.2.0-beta.1` in your Prisma app to see if it works fine?

Reply:
> Yes, I just commited (and merged) the two 'regrets' in case-insensitive.

:rocket: I'm not sure if it's more performant or not, but it's worth it to keep the 'regrets' more consistent (and I'm sure any performance difference would be tiny in comparison to how long it takes to encrypt/decrypt stuff!).

> Can you try and `patch@47ng/cloak@1.2.0-beta.1` in your Prisma app to see if it works fine? Then I can publish the release on `latest`.

Confirmed working!

Sorry for the late response. I updated from Ubuntu 22.04 to 24.04 this weekend, and as part of it, I replaced my Docker environment with Podman, so I had to re-download all my Prisma containers!

I did have to patch `prisma-field-encryption` with:

```diff
diff --git a/dist/encryption.js b/dist/encryption.js
index 99f9cd6ff2e0bd6a721ff589e4c37811183db986..4c83b9927b534a3e68b20e11ed05eb0249a2e5be 100644
--- a/dist/encryption.js
+++ b/dist/encryption.js
@@ -104,7 +104,7 @@ function decryptOnRead(params, result, keys, models, operation) {
     const fatalDecryptionErrors = [];
     (0, visitor_1.visitOutputTargetFields)(params, result, models, function decryptFieldValue({ fieldConfig, value: cipherText, path, model, field }) {
         try {
-            if (!cloak_1.cloakedStringRegex.test(cipherText)) {
+            if (!cloak_1.parseCloakedString(cipherText)) {
                 return;
             }
             const decryptionKey = (0, cloak_1.findKeyForMessage)(cipherText, keys.keychain);
```

(should be the equivalent of my patch in https://github.com/47ng/prisma-field-encryption/issues/122, except on the generated `dist/`)

Reply:
Done: https://github.com/47ng/prisma-field-encryption/pull/123

Making an integration test took a bit longer! It looks like without a `?mode=strict`, the errors are ignored.